src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java

```diff
@@ -0,0 +1,133 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.destroystokyo.paper.event.entity.CreeperIgniteEvent;
+import com.google.common.base.Preconditions;
+import org.bukkit.entity.Creeper;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.entity.CreeperPowerEvent;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public class CreeperMock extends MonsterMock implements Creeper
+{
+
+	private boolean powered = false;
+	private int maxFuseTicks = 30;
+	private int fuseTicks = 0;
+	private int explosionRadius = 3;
+	private boolean ignited = false;
+
+	public CreeperMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public boolean isPowered()
+	{
+		return this.powered;
+	}
+
+	@Override
+	public void setPowered(boolean value)
+	{
+		CreeperPowerEvent.PowerCause cause = powered ? CreeperPowerEvent.PowerCause.SET_ON : CreeperPowerEvent.PowerCause.SET_OFF;
+
+		if (!this.callPowerEvent(cause))
+		{
+			this.powered = value;
+		}
+	}
+
+	private boolean callPowerEvent(CreeperPowerEvent.PowerCause cause)
+	{
+		CreeperPowerEvent event = new CreeperPowerEvent( this, cause);
+		this.getServer().getPluginManager().callEvent(event);
+		return event.isCancelled();
+	}
```

Comment:
```suggestion
		if (!new CreeperPowerEvent(this, cause).callEvent())
		{
			this.powered = value;
		}
	}
```

Reply:
This is how it's handled in CraftBukkit, thats why I think we should keep it

Reply:
It's only used in one place. We don't need to copy CraftBukkit's code 1 for 1, just as long as the functionality is the same. If it needs to be used in more places in the future, then a method would be good. But for now, doing it this way is much cleaner.